src/wallet/rpcwallet.cpp

```diff
@@ -2645,10 +2645,26 @@ UniValue listwallets(const JSONRPCRequest& request)
 
         LOCK(pwallet->cs_wallet);
 
-        obj.push_back(pwallet->GetName());
+        loaded.push_back(pwallet->GetName());
     }
 
-    return obj;
+    UniValue available(UniValue::VARR);
+    fs::path walletdir = GetDataDir();
+    const unsigned char bdb_magic[4] = {0x62, 0x31, 0x05, 0x00}; // Berkeley DB Btree magic bytes
+    for (fs::directory_iterator it(walletdir); it != fs::directory_iterator(); ++it) {
+        std::ifstream file(it->path().string(), std::ios::binary);
+        file.seekg(12, std::ios::beg); // Magic bytes start at offset 12
+        unsigned char fileBytes[4] = {0};
+        file.read((char*)fileBytes, sizeof(fileBytes)); // Read 4 bytes of file to compare against magic
+        if (is_regular_file(*it) && file.is_open() && memcmp(fileBytes, bdb_magic, sizeof(bdb_magic)) == 0) {
```

Comment:
Please explicitly use `fs::is_regular_file()` (this only compiles because of Argument Dependant Lookups http://en.cppreference.com/w/cpp/language/adl which makes it difficult to understand where this function is coming from).

`boost::filesystem::is_regular_file()` will return true if called on a symlink to a regular file. We don't want those to show up in the available list since they can't be loaded as a wallet, so you should extend this if test with `fs::is_symlink()`.